base/ubi9/entrypoint.sh

```diff
@@ -31,6 +31,64 @@ if ! whoami &> /dev/null; then
     echo "${USER_NAME:-user}:x:$(id -u):0:${USER_NAME:-user} user:${HOME}:/bin/bash" >> /etc/passwd
     echo "${USER_NAME:-user}:x:$(id -u):" >> /etc/group
   fi
+
+  # The user namespace is created when `UserNamespacesSupport` feature is enabled and `hostUsers` is set to false in Pod spec.
+  # See for more details:
+  # - https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#introduction
+  # Assume, that HOST_USERS environment variable is provided and is set to false in that case.
+  # If so, update /etc/sub*id files to reflect the valid UID/GID range.
+  if [ "${HOST_USERS}" == "false" ]; then
+    echo "Running in a user namespace"
+    echo "User information: $(id)"
+    if [ -f /proc/self/uid_map ]; then
+      # Typical output of `/proc/self/uid_map`:
+      # 1. When NOT running in a user namespace:
+      #         0          0 4294967295
+      # 2. When running in a user namespace:
+      #         0 1481179136      65536
+      # 3. When container is run in a user namespace:
+      #         0       1000          1
+      #         1       1001      64535
+      # We can use the content of /proc/self/uid_map to detect if we are running in a user namespace.
+      # However, to keep things simple, we will rely on HOST_USERS environment variable only.
+      # This way, we avoid breaking anything.
+      echo "/proc/self/uid_map content: $(cat /proc/self/uid_map)"
+    fi
+
+    # Why do we need to update /etc/sub*id files?
+    # From one hand, we are already in user namespace, so we don't have more than 65536 UIDs/GIDs available.
+    # See for more details:
+    # - https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#id-count-for-each-of-pods
+    # From the other hand, podman creates a new user namespace for a container being launched and maps the container's user
+    # and group IDs (UID/GID) to the corresponding user on the current namespace.
+    # For the mapping, podman refers to the /etc/sub*id files.
+    # See for more details:
+    # - https://man7.org/linux/man-pages/man5/subuid.5.html
+    # So, if user ID exceeds 65536 number, than it can be not be mapped, since it is not just available.
+    # As a result, podman commands fails.
+
+    # Even though, the range can be extended if needed by configuration, we can relay on the fact that IDs number at
+    # least 65536 are always available in a user namespace.
+    # See for more details:
+    # - https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-UserNamespaces
+
+    # To ensure the provided user ID stays within this range, `runAsUser` in the Pod spec should be set to a value below 65536.
+    # In OpenShift, the Container Security Context Constraint (SCC) should be created accordingly.
+    if [ "$(id -u)" -lt 65536 ]; then
+      START_ID=$(( $(id -u) + 1 ))
+      END_ID=$(( 65536 - ${START_ID} ))
```

Comment:
Because of wording in https://man7.org/linux/man-pages/man5/subuid.5.html, could we rename to COUNT or some other name emphasizing that it is a count?

```suggestion
      COUNT=$(( 65536 - ${START_ID} ))
```